conan/tools/cmake/toolchain.py

```diff
@@ -513,6 +526,38 @@ def _get_generator_platform(self, generator):
                     "armv8": "ARM64"}.get(arch)
         return None
 
+    def _get_cross_build(self):
+        system_name = self._conanfile.conf["tools.cmake.cmaketoolchain:system_name"]
+        system_version = self._conanfile.conf["tools.cmake.cmaketoolchain:system_version"]
+        system_processor = self._conanfile.conf["tools.cmake.cmaketoolchain:system_processor"]
+
+        settings = self._conanfile.settings
+        if hasattr(self._conanfile, "settings_build"):
+            os_ = settings.get_safe("os")
+            arch = settings.get_safe("arch")
+            settings_build = self._conanfile.settings_build
+            os_build = settings_build.get_safe("os")
+            arch_build = settings_build.get_safe("arch")
+
+            if system_name is None:  # Try to deduce
+                if os_ not in ('Macos', 'iOS', 'watchOS', 'tvOS'):  # Handled by AppleBlock
+                    if os != os_build:
```

Comment:
```suggestion
                    if os_ != os_build:
```

Reply:
Good catch, I have fixed it, added an extra check and a new test to cover it.